include/NAS2D/Resources/Sprite.h

```diff
@@ -136,20 +128,18 @@ class Sprite
 		int frameDelay() const { return mFrameDelay; }
 
 	private:
-		std::string		mSheetId;
-
-		int				mFrameDelay;
-		int				mAnchorX, mAnchorY;
-
-		Rectangle_2d	mRect;
+		std::string mSheetId{};
```

Comment:
I'm kind of on the fence about these empty initializers. Seems simpler to not have them there at all. If they weren't being initialized before, they probably don't need to be initialized now.

Reply:
Not initializing variables is a source of bugs. "Empty initializers" as you call them are actually zero initializers and are a good thing. They automatically zero the memory for `struct`-like types, or call the default constructor for non-trivial types, instead of initializing to whatever garbage was there already.

Reply:
I'm aware of the difference. I'm suggest zero initializing may not be needed here. If it's not needed why waste the cycles.

As for avoiding bugs, there may be a deeper issue here. Why was no initialization needed before. That may be an indication the variable shouldn't be a class member. Perhaps it would fit better as a local variable in member functions. There were other places in the code where such a change was made.

I would like to look into removing uninitialized member variables before considering default initializing them.

Reply:
I'm looking back at this now, and realizing part of what bothered me about it. Many of the objects being initialized are not primitive types. That means default initialization using the class constructor. It does not simply zero initialize, as may be done for primitive types.

I think I would prefer omitting the `{}` for object types that have a non-trivial default constructor, such as `std::string` and `std::vector`. Doubly so for `std::vector` since `{}` implies list initialization, which is not really the same as default initialization (just the result is the same in this case for the empty list).

----

For primitive types, if there was no previous initialization, I would assume non-trivial initialization is done in the constructor. If no initialization is done there, it's likely a bug. Using default initialization for those values may hide such bugs.

This reminds me of an article I read a few months back. It mentioned the once common practice of clearing pointer values after calling `delete`. In good code it's an inefficient wasted operation. In bad code, it addresses the symptoms of trying to double `delete`, rather than address the actual cause of the double `delete` itself. The author suggested not bothering with this practice. Just having the code crash may actually be a good thing, since it alerts you to the bug.

I think similar reasoning applies here. A crash, particularly in development, may be preferable. It alerts you to a bug so it can be fixed, rather than papering over the symptoms. A default initialized value might be enough to avoid a spectacular crash, though it still might not work correctly. In such cases, default initialization addresses the symptom, not the cause.

----

For a concrete example, the `Image` copy constructor doesn't set the `_center` value. Default initializing it to `{0, 0}` may make things work smoothly, but its still not correct. (Issue #402).